core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -646,16 +677,37 @@ private[chisel3] object Builder extends LazyLogging {
     }
   }
 
+  // Builds a RenameMap for all Views that do not correspond to a single Data
+  // These Data give a fake ReferenceTarget for .toTarget and .toReferenceTarget that the returned
+  // RenameMap can split into the constituent parts
+  private[chisel3] def makeViewRenameMap: RenameMap = {
```

Comment:
Daaang I didn't realize you did this. Very nice.

Reply:
Yeah, it worked out a lot better than I expected. This is the sort of thing that weird, custom Stages might run afoul of so it's a thing to mention in release notes, but it's actually a fairly elegant solution in my opinion.